CMakeLists.txt

```diff
@@ -54,56 +50,82 @@ set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
+            "Consider adding Qt to the CMAKE_PREFIX_PATH,\n" 
+            "i.e. start cmake with the option -DCMAKE_PREFIX_PATH=\"/path/to/qt\"")
+    endif()
+endif()
+
 include(cmake/globalconfig.cmake)
 
-#--------------------------------------------------------------------
 # Application setup
-option(IVW_APP_QTBASE       "Build base for Qt apps. Used by the Qt network editor and the Tiny Qt app" ON)
 option(IVW_APP_INVIWO       "Build Inviwo Qt network editor application" ON)
 option(IVW_APP_MINIMAL_GLFW "Build Inviwo Tiny GLFW Application" OFF)
-option(IVW_APP_MINIMAL_QT   "Build Inviwo Tiny QT Application" OFF)
+option(IVW_APP_MINIMAL_QT   "Build Inviwo Tiny QT Application" ON)
 option(IVW_APP_PYTHON       "Build Inviwo Python Application" ON)
 
-if((IVW_APP_INVIWO OR IVW_APP_MINIMAL_QT OR IVW_APP_PYTHON) AND NOT IVW_APP_QTBASE)
-    set(IVW_APP_QTBASE ON CACHE BOOL 
-        "Build base for qt applications. Used by Qt network editor and Qt minimal application" 
-        FORCE
-    )
-    message(STATUS "IVW_APP_QTBASE was set to build, due to dependencies")
-endif()
-
 ivw_enable_modules_if(IVW_APP_INVIWO QtWidgets)
 ivw_enable_modules_if(IVW_APP_MINIMAL_QT QtWidgets)
 ivw_enable_modules_if(IVW_APP_MINIMAL_GLFW GLFW)
 ivw_enable_modules_if(IVW_APP_INVIWO_DOME SGCT)
-if(IVW_USE_PYTHON)
-    ivw_enable_modules_if(IVW_APP_PYTHON Python3 Python3Qt QtWidgets)
-endif()
+ivw_enable_modules_if(IVW_APP_PYTHON Python3 Python3Qt QtWidgets)
 
 option(IVW_TEST_INTEGRATION_TESTS "Build inviwo integration test" ON)
 ivw_enable_modules_if(IVW_TEST_INTEGRATION_TESTS GLFW Base)
 
-# Try to find qt and add it if it is not already in CMAKE_PREFIX_PATH
-if(NOT "${CMAKE_PREFIX_PATH}" MATCHES "[Qq][Tt]")
-    include(cmake/locateqt.cmake)
-    ivw_locateqt(QT_PATH)
-    if(QT_PATH)
-        ivw_debug_message(STATUS "Found qt at: ${QT_PATH}")
-        list(APPEND CMAKE_PREFIX_PATH ${QT5_PATH})
-    endif()
-endif()
-
 add_custom_target(uninstall COMMENT "Dummy target to prevent other uninstalls")
 
 add_subdirectory(ext)                        # Add external libraries.
 add_subdirectory(tools/meta)                 # Add inviwo meta library, used in the qt editor
 add_subdirectory(tools/tracy)                # Add tracy wrapper, used for profiling
 if(IVW_TEST_INTEGRATION_TESTS OR IVW_TEST_UNIT_TESTS)
     add_subdirectory(tests/testutil)         # Add test utils, used in the module unit tests
-endif()                    
+endif()
+
 ivw_register_modules(all_modules)            # Add inviwo modules
-add_subdirectory(src/qt)                     # Add the Qt network editor
-add_subdirectory(apps)                       # Add all the applications, uses the modules.
+
+# Add all the applications, uses the modules.
+if(IVW_ENABLE_QT)
+    add_subdirectory(src/qt/applicationbase)
+    if(IVW_APP_INVIWO)
+        add_subdirectory(src/qt/editor)
+        add_subdirectory(apps/inviwo)
+    endif()
+    if(IVW_APP_PYTHON)
```

Comment:
shouldn't that be depending on Python as well?

Reply:
No. I you say that you want python, then not getting it should be an error.